contracts/upgradeable_contracts/ChaiConnector.sol

```diff
@@ -0,0 +1,156 @@
+    * @param amount Amount of Chai tokens to transfer
+    */
+    function payInterest(address recipient, uint256 amount) external onlyOwner {
+        // check that the remaining chai balance will be sufficient to cover all invested DAI tokens
+        require(chaiBalance() - amount >= investedAmountInChai());
+        chaiToken().transfer(recipient, amount);
+
+        require(dsrBalance() >= investedAmountInDai());
+    }
+
+    /**
+    * @dev Returns current used chai contract
+    * @return chai contract address
+    */
+    function chaiToken() public view returns (IChai) {
+        return IChai(addressStorage[CHAI_TOKEN]);
+    }
+
+    /**
+    * @dev Evaluates bridge balance for tokens, holded in DSR
+    * @return Balance in dai, truncated
+    */
+    function dsrBalance() public view returns (uint256) {
+        uint256 chi = chaiToken().pot().chi();
+        return rmul(chi, chaiBalance());
+    }
+
+    /**
+    * @dev Evaluates bridge balance in Chai tokens
+    * @return Balance in chai, exact
+    */
+    function chaiBalance() public view returns (uint256) {
+        return chaiToken().balanceOf(address(this));
+    }
+
+    /**
+    * @dev Evaluates exact current invested amount, id DAI
+    * @return Value in DAI
+    */
+    function investedAmountInDai() public view returns (uint256) {
+        return uintStorage[INVESTED_AMOUNT];
+    }
+
+    function setInvestedAmointInDai(uint256 amount) internal {
+        uintStorage[INVESTED_AMOUNT] = amount;
+    }
+
+    /**
+    * @dev Evaluates amount of chai tokens that is sufficent to cover 100% of the invested DAI
+    * @return Amount in chai, ceiled
+    */
+    function investedAmountInChai() internal returns (uint256) {
+        IPot pot = chaiToken().pot();
+        // solhint-disable-next-line not-rely-on-time
+        uint256 chi = (now > pot.rho()) ? pot.drip() : pot.chi();
+        return rdivup(investedAmountInDai(), chi);
+    }
+
+    /**
+    * @dev Invests DAI into Chai
+    * @param amount Amount of DAI to invest
+    */
+    function _convertDaiToChai(uint256 amount) internal {
+        setInvestedAmointInDai(investedAmountInDai() + amount);
+        erc20token().approve(chaiToken(), amount);
+        chaiToken().join(address(this), amount);
+    }
+
+    /**
+    * @dev Redeems DAI from Chai, the total redeemed amount will be at least equal to specified amount
+    * @param amount Amount of DAI to redeem
+    */
+    function _convertChaiToDai(uint256 amount) internal {
+        uint256 invested = investedAmountInDai();
+        if (amount >= invested) {
+            chaiToken().draw(address(this), invested);
+            setInvestedAmointInDai(0);
+        } else if (amount > 0) {
+            uint256 initialDaiBalance = erc20token().balanceOf(address(this));
+            chaiToken().draw(address(this), amount);
+            uint256 redeemed = erc20token().balanceOf(address(this)) - initialDaiBalance;
```

Comment:
```suggestion
            uint256 redeemed = erc20token().balanceOf(address(this)).sub(initialDaiBalance);
```

Reply:
Here, SafeMath looks redundant as well. Balance after `draw` cannot be less than the initial one.